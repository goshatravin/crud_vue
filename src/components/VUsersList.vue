<template>
    <div class="home">
        <div class="new-user">
            <button v-on:click='addNewUser'>New user</button>
        </div>
        <div class="all-users">
            <div class="new_one" v-for="(item, index) of usersArray"
            :key='item.name'
            >
                <p>{{item.name}}</p>
                <p>{{item.position}}</p>
                <p>{{item.address}}</p>
                <p>{{item.company}}</p>
                <p :name='index' v-on:click="edit">EDIT</p>
                <p>DEL</p>
            </div>
        </div>
    </div>
</template>

<script>
import { mapState } from 'vuex';

export default {
    name: 'VUsersList',
    methods: {
        // Show pop-up with registration form for registration new users
        addNewUser() {
            this.$store.dispatch('TOGGLE_POP_UP', true);
        },
        // eslint-disable-next-line no-unused-vars
        edit(e, target) {
            const myObj = {
                data: this.usersArray[e.target.getAttribute('name')],
                id: e.target.getAttribute('name'),
            };
            this.$store.dispatch('EDIT_USER', myObj);
        },
    },
    computed: {
        ...mapState({
            usersArray: (state) => state.newUser.usersArray,
        }),
    },
};
</script>

<style>
.new_one{
    border: 1px solid black;
}
</style>
