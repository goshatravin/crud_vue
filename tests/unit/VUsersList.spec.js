import { shallowMount } from '@vue/test-utils';
import VUsersList from '@/components/VUsersList.vue';

describe('VUsersList', () => {
    const build = () => {
        const wrapper = shallowMount(VUsersList);

        return {
            wrapper,
            regForm: () => wrapper.find('.new-user'),
        };
    };

    it('Renders the component', () => {
        // arrange
        const { wrapper } = build();
        // assert
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('Renders main components', () => {
        const { regForm } = build();
        expect(regForm().exists()).toBe(true);
    });
});
