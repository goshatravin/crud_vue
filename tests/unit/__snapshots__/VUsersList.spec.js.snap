// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VUsersList Renders the component 1`] = `
<div class="home">
  <div class="new-user"><button></button></div>
</div>
`;
